src/components/Canvas/CanvasTemplate.vue

```diff
@@ -234,12 +275,88 @@ export default {
             top: newTopPosition,
             behavior: "smooth",
           });
+          this.insertionManagement(imageId, currentImageTopPosition, currentImageLeftPosition, imagePosition);
         }, 200);
       } else {
         focusElement.hasFocus = false;
         this.currentFocus.push(false);
       }
     },
+    // This method analyzes the state of the canvas and insert the new block when and where it's necessary
+    insertionManagement(imageId, currentImageTopPosition, currentImageLeftPosition, imagePosition) {
+      // First, check if it's necessary to do something.
+      if (shouldInsert(imageId, this.imageBlockController)) {
```

Comment:
```suggestion
      if (!shouldInsert(imageId, this.imageBlockController)) {
      return ;
      }
```
better, else we are looking for an else